src/include/buffer.h

```diff
@@ -411,22 +412,32 @@ namespace buffer CEPH_BUFFER_API {
     void push_front(ptr& bp) {
       if (bp.length() == 0)
 	return;
+      _len += bp.length();
       _buffers.push_front(bp);
+    }
+    void push_front(ptr&& bp) {
+      if (bp.length() == 0)
+	return;
+      _buffers.push_front(std::move(bp));
```

Comment:
need to call bp.length() before push_front() here. it looks like you applied this fix to `void push_front(ptr& bp) {` instead